src/test/java/com/meilisearch/integration/DocumentsTest.java

```diff
@@ -78,5 +77,143 @@ public void testAddDocumentsMultiple() throws Exception {
+		assertEquals(20, movies.length);
+		for (int i = 0; i < movies.length; i++) {
+			assertEquals(movies[i].getTitle(), testData.getData().get(i).getTitle());
+			assertEquals(movies[i].getId(), testData.getData().get(i).getId());
+			String[] expectedGenres = testData.getData().get(i).getGenres();
+			String[] foundGenres = movies[i].getGenres();
+			for (int x = 0; x < expectedGenres.length; x++) {
+				assertEquals(expectedGenres[x], foundGenres[x]);
+			}
+		}
+	}
+
+	/**
+	 * Test GetDocuments with limit
+	 */
+	@Test
+	public void testGetDocumentsLimit() throws Exception {
+
+		String indexUid = "GetDocumentsLimit";
+		int limit = 24;
+		client.createIndex(indexUid);
+		Index index = client.getIndex(indexUid);
+
+		TestData<Movie> testData = this.getTestData(MOVIES_INDEX, Movie.class);
+		UpdateStatus updateInfo = this.gson.fromJson(
+			index.addDocuments(testData.getRaw()),
+			UpdateStatus.class
+		);
+
+		index.waitForPendingUpdate(updateInfo.getUpdateId());
+		Movie[] movies = this.gson.fromJson(index.getDocuments(limit), Movie[].class);
+		assertEquals(limit, movies.length);
+		for (int i = 0; i < movies.length; i++) {
+			assertEquals(movies[i].getId(), testData.getData().get(i).getId());
+		}
+	}
+
+	/**
+	 * Test deleteDocument
+	 */
+	@Test
+	public void testDeleteDocument() throws Exception {
+
+		String indexUid = "DeleteDocument";
+		client.createIndex(indexUid);
+		Index index = client.getIndex(indexUid);
+
+		TestData<Movie> testData = this.getTestData(MOVIES_INDEX, Movie.class);
+		UpdateStatus updateInfo = this.gson.fromJson(
+			index.addDocuments(testData.getRaw()),
+			UpdateStatus.class
+		);
+		index.waitForPendingUpdate(updateInfo.getUpdateId());
+
+		Movie[] movies = this.gson.fromJson(index.getDocuments(), Movie[].class);
+		Movie toDelete = movies[0];
+		updateInfo = this.gson.fromJson(
+			index.deleteDocument(toDelete.getId()),
+			UpdateStatus.class
+		);
+		index.waitForPendingUpdate(updateInfo.getUpdateId());
+
+		assertThrows(
+			Exception.class,
+			() -> index.getDocument(toDelete.getId())
+		);
+	}
+
+	/**
+	 * Test deleteDocuments
+	 */
+	@Test
+	public void testDeleteDocuments() throws Exception {
+
+		String indexUid = "DeleteDocuments";
+		client.createIndex(indexUid);
+		Index index = client.getIndex(indexUid);
+
+		TestData<Movie> testData = this.getTestData(MOVIES_INDEX, Movie.class);
+		UpdateStatus updateInfo = this.gson.fromJson(
+			index.addDocuments(testData.getRaw()),
+			UpdateStatus.class
+		);
+		index.waitForPendingUpdate(updateInfo.getUpdateId());
+
+		Movie[] movies = this.gson.fromJson(index.getDocuments(), Movie[].class);
+		assertEquals(20, movies.length);
+
+		updateInfo = this.gson.fromJson(
+			index.deleteDocuments(),
+			UpdateStatus.class
+		);
+		index.waitForPendingUpdate(updateInfo.getUpdateId());
+
+		movies = this.gson.fromJson(index.getDocuments(), Movie[].class);
+		assertEquals(0, movies.length);
+	}
+
```

Comment:
There is a missing test for documents deletion. I don't know if `deleteDocuments` is the [delete-batch route](https://docs.meilisearch.com/references/documents.html#delete-documents) or the [delete-all one](https://docs.meilisearch.com/references/documents.html#delete-documents) but there is one missing in the tests.

Reply:
Both links point to the same place 😄 
But I created an issue https://github.com/meilisearch/meilisearch-java/issues/42

Reply:
Haha my bad! Your comment and the issue answer perfectly my question!

Reply:
No, it was you who inspired the issue creation 🎉